framework/include/userobject/ElementUserObject.h

```diff
@@ -63,12 +63,20 @@ class ElementUserObject :
   virtual void execute() = 0;
 
   /**
+   * This function will be called with the shape functions for jvar initialized. It
+   * can be used to compute Jacobian contributions of the UserObject.
+   */
+  virtual void executeJacobian(unsigned int /*jvar*/) {}
+
+  /**
    * Must override.
    *
    * @param uo The UserObject to be joined into _this_ object.  Take the data from the uo object and "add" it into the data for this object.
    */
   virtual void threadJoin(const UserObject & uo) = 0;
 
+  bool requestedJacobian(unsigned int var) const;
```

Comment:
Yes... with the caveat that we don't even need a list in Assembly (see my comment back over on the PR).